src/process.cr

```diff
@@ -440,6 +443,79 @@ class Process
       abort("Unresolvable state, exiting...")
     end
   end
+
+  # Sets the real, effective, and saved `User` to the one specified.
+  def self.user=(user : System::User) : Nil
+    return if LibC.setuid(user.uid) == 0
+    raise Errno.new("The calling process was not privileged")
+  end
+
+  # Sets the real, effective, and saved `Group` to the one specified.
+  def self.group=(group : System::Group) : Nil
+    return if LibC.setgid(group.gid) == 0
+    raise Errno.new("The calling process was not privileged")
+  end
+
+  # Sets the real, effective, and saved `User` and `Group` to the ones specified.
+  def self.become(user : System::User, group : System::Group) : Nil
+    self.group = group
+    self.user = user
+  end
+
+  # Returns a `Bool` indicating if the current process is running as root.
+  def self.root? : Bool
```

Comment:
Not sure it's really useful, we already have `#zero?`. Most of us handling with users knows that uid 0 is root.

Reply:
> Most of us handling with users knows that uid 0 is root.
That depends on the operating system.

Reply:
For example, there is `toor` on BSD that is also uid 0.
It would make more sense to call the administrator to a more generic name like `admin?`.

Reply:
I don't think so. This method should indicate if the process is running as `root`. That's a specific property of any process on almost anything other than windows.